contrib/hashicorp/consul/option.go

```diff
@@ -20,6 +22,8 @@ type clientConfig struct {
 	serviceName   string
 	operationName string
 	analyticsRate float64
+	hostname      string
+	consulConfig  *consul.Config
```

Comment:
AFAICT, consulConfig is not actually used once we set `hostname`, so we should drop it.
It's easy to add back in later if necessary if we need to get some info out of it.